### PR TITLE
Alter field url on webhook

### PR DESCRIPTION
**Pull request by santos22**
See https://github.com/readthedocs/readthedocs.org/issues/6625

~~Went with the option to explicitly ask users to use http or https schemes~~

Removed custom URL validator in favor of the built-in Django one.

Ran migration with the following command:
```
 python manage.py makemigrations --name webhook_url_set_blank_default projects
```

Also cleaned up some class names that were misspelled!

### CODE REVIEW

File: readthedocs/projects/forms.py [hunk 1]
```diff
@@ -473,6 +473,10 @@ def clean_url(self):
             raise forms.ValidationError(
                 _('This field is required.')
             )
+        if 'https' not in url:
```
Comment (santos22) [inline readthedocs/projects/forms.py:476]: Other options include:
```python
# option 1
if url.startswith('http://'):
    raise forms.ValidationError(
            _('Please provide an https URL.')
        )

# option 2
from urllib.parse import urlparse

url = urlparse(url)

if url.scheme.startswith('http://'):
    raise forms.ValidationError(
            _('Please provide an https URL.')
        )
```

Comment (stsewd) [reply readthedocs/projects/forms.py:476]: I think is better if we check for `^https?://` with a regex

File: readthedocs/rtd_tests/tests/test_project_forms.py [hunk 2]
```diff
@@ -710,6 +710,14 @@ def test_wrong_inputs_in_webhookform(self):
         self.assertDictEqual(form.errors, {'url': ['Enter a valid URL.']})
         self.assertEqual(self.project.webhook_notifications.all().count(), 0)
 
+        data = {
+            'url': 'http://www.example.com/'
```
Comment (stsewd) [inline readthedocs/rtd_tests/tests/test_project_forms.py:714]: We should add more tests cases here. Like urls that have http in the middle, end, don't start with a protocol, etc

File: readthedocs/projects/forms.py (hunk 1 above)
Comment (humitos) [reply readthedocs/projects/forms.py:476]: Probably better to use `urlparse` and check for the scheme than applying a regex over a user input.

File: readthedocs/rtd_tests/tests/test_project_forms.py (hunk 2 above)
Comment (santos22) [reply readthedocs/rtd_tests/tests/test_project_forms.py:714]: Will add more tests 👍

File: readthedocs/projects/forms.py [hunk 3]
```diff
@@ -473,6 +473,12 @@ def clean_url(self):
             raise forms.ValidationError(
                 _('This field is required.')
             )
+        parsed_url = urlparse(url)
+
+        if parsed_url.scheme != 'https':
```
Comment (santos22) [inline readthedocs/projects/forms.py:478]: I can also change this to look for other schemes like 'ftp' and 'ftps'.

Comment (stsewd) [reply readthedocs/projects/forms.py:478]: Let's keep the `http` protocol. `not in ('http', 'https')`

File: readthedocs/projects/models.py [hunk 4]
```diff
@@ -1447,7 +1447,6 @@ def __str__(self):
 class WebHook(Notification):
     url = models.URLField(
         max_length=600,
-        blank=True,
```
Comment (stsewd) [inline readthedocs/projects/models.py:1450]: This change will require a migration. `manage.py migrate --name make_url_field_not_blank projects` or similar.

Comment (santos22) [reply readthedocs/projects/models.py:1450]: Good catch 👍

**Pull request merged**